templates/cluster-template.yaml

```diff
@@ -0,0 +1,200 @@
+---
```

Comment:
There really isn't anything that I can tell different between the -high-availability.yaml template and this one, other than this one hardcodes the replicas on the KubeadmControlPlane, I think you can likely just expose replicas here and not have a separate high-availability template.

Reply:
Problem is that the ControlPlaneEndpoint is configured differently. In one case we're using Octavia to create a LB in the other case just a floating ip on the machine

HA case:
OpenStackCluster:
```
 managedAPIServerLoadBalancer: true
  apiServerLoadBalancerFloatingIP: ${OPENSTACK_CONTROLPLANE_IP}
  apiServerLoadBalancerPort: 6443
  apiServerLoadBalancerAdditionalPorts:
  - 22
  useOctavia: true
```

Single-node master:
OpenStackCluster
```
controlPlaneEndpoint:
    host: ${OPENSTACK_CONTROLPLANE_IP}
    port: 6443
useOctavia: false
```

OpenStackMachineTemplate: (${CLUSTER_NAME}-control-plane)
```
floatingIP: ${OPENSTACK_CONTROLPLANE_IP}
```

The single node  variant is mostly there because not everyone has Octavia (or rather LBaaS) support.

So maybe the naming is just wrong. Actually it's one variant with LBaaS (with Neutron lbaasv2 or Octavia) and one variant with Floating IP on the one master node instead of an LB.

Reply:
I'll think this through later and probably rename the flavors, e.g.
* default => with LBaaS (Octavia or Neutron LBaaS v2)
* -without-lbaas => single node master without lbaas (maybe I'll find a better name for it)

Reply:
I changed it as described and called the non-default flavor "without-lb"

@jichenjc You have to use the other flavor now in case you're testing without lbaas